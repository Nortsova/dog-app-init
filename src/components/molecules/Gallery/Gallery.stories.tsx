import type { Meta, StoryObj } from '@storybook/react';
import { Gallery } from '.';

const meta = {
  title: 'Molecules/Gallery',
  component: Gallery
} satisfies Meta<typeof Gallery>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {}
};
