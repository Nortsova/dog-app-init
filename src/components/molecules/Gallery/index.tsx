import { Box, Button, IconButton, Typography, styled } from '@mui/material';
import { useState } from 'react';
import { CardItem } from '../../atoms/CardItem';
import { IconHeart } from '../../atoms/IconHeart';
import { useGetBreedsQuery } from '../../../services/breeds';

const StyledBox = styled(Box)(({ theme }) => ({
  display: 'grid',
  gridTemplateColumns: 'repeat(3, 372px)',
  gridGap: 22,
  '& img': {
    marginBottom: '14px',
    height: '274px'
  },
  '& p': {
    padding: '0px 28px'
  },
  '& .MuiButton-text': {
    position: 'absolute',
    bottom: '20px',
    right: '20px',
    display: 'block',
    backgroundColor: 'transparent',
    color: theme.palette.warning.main
  }
}));

export const Gallery = () => {
  const { data: breeds, isLoading } = useGetBreedsQuery({ limit: 6, page: 0 });
  const [isHoveredCard, setIsHoveredCard] = useState<number | null>(null);

  return (
    <>
      {isLoading && <div>Loading...</div>}
      <StyledBox>
        {(breeds || []).map((item, index) => {
          const isHover = isHoveredCard === index;

          return (
            <CardItem
              key={item.id}
              variant="primary"
              borderRadius={0}
              sx={{ paddingBottom: '29px' }}
              onMouseOver={() => setIsHoveredCard(index)}
              onMouseOut={() => setIsHoveredCard(null)}
            >
              <img src={item.image.url} alt={item.name} />
              <IconButton
                sx={{ position: 'absolute', top: '10px', right: '10px' }}
              >
                {isHover ? (
                  <IconHeart state="hover" />
                ) : (
                  <IconHeart state="default" />
                )}
              </IconButton>
              <Typography fontSize={26} color="#5F556D">
                {item.name}
              </Typography>
              <Typography>
                <Typography component="span" fontSize={20} color="#5F556D">
                  Temperament:{' '}
                </Typography>
                <Typography component="span" fontSize={20} color="#ADA7B8">
                  {item.temperament}
                </Typography>
              </Typography>
              {isHover && (
                <Button variant="text" color="inherit">
                  more
                </Button>
              )}
            </CardItem>
          );
        })}
      </StyledBox>
    </>
  );
};
