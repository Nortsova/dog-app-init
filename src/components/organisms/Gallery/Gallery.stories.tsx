import type { Meta, StoryObj } from '@storybook/react';
import { Gallery } from '.';

const meta = {
  title: 'Organisms/Gallery',
  component: Gallery
} satisfies Meta<typeof Gallery>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Main: Story = {
  args: {
    info: {
      title: 'Title'
    }
  }
};
