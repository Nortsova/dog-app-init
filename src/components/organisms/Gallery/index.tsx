import { Box, Stack } from '@mui/material';
import { useState } from 'react';
import { GalleryScreen } from '../../molecules/GalleryScreen';
import { GalleryList } from '../../molecules/GalleryList';
import { GalleryItemProps } from '../../../types/gallery';

const galleryItems: GalleryItemProps[] = [
  {
    title: 'Image 1',
    url: 'https://loremflickr.com/640/361',
    id: '1'
  },
  {
    title: 'Image 2',
    url: 'https://loremflickr.com/640/362',
    id: '2'
  },
  {
    title: 'Image 3',
    url: 'https://loremflickr.com/640/363',
    id: '3'
  },
  {
    title: 'Image 4',
    url: 'https://loremflickr.com/640/364',
    id: '4'
  }
];

export const Gallery = () => {
  const [selectedImage, setSelectedImage] = useState<GalleryItemProps>(
    galleryItems[0]
  );
  const handleClick = (newImage: GalleryItemProps) => () => {
    setSelectedImage(newImage);
  };
  return (
    <Stack direction="row">
      <Box
        sx={{
          maxWidth: '500px'
        }}
      >
        <GalleryScreen title={selectedImage.title} url={selectedImage.url} />
      </Box>
      <GalleryList items={galleryItems} handleClick={handleClick} />
    </Stack>
  );
};
