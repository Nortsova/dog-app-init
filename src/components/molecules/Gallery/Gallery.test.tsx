import { composeStories } from '@storybook/react';
import * as stories from './Gallery.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('Gallery', () => {
  const { Primary } = composeStories(stories);
  testSnapshot(<Primary />);
});
