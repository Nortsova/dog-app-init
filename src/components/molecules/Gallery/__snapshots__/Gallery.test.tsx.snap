// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery should match snapshot Component 1`] = `
.emotion-0 {
  display: grid;
  grid-template-columns: repeat(3, 372px);
  grid-gap: 22px;
}

.emotion-0 img {
  margin-bottom: 14px;
  height: 274px;
}

.emotion-0 p {
  padding: 0px 28px;
}

.emotion-0 .MuiButton-text {
  position: absolute;
  bottom: 20px;
  right: 20px;
  display: block;
  background-color: transparent;
  color: #FFCF32;
}

<div>
  <div>
    Loading...
  </div>
  <div
    class="MuiBox-root emotion-0"
  />
</div>
`;
